src/users/__tests__/userHelpers-test.js

```diff
@@ -259,22 +257,23 @@ describe('filterUserStartWith', () => {
     const expectedUsers = [user1, user3];
     expect(filterUserStartWith(users, 'app', selfUser.user_id)).toEqual(expectedUsers);
   });
-});
-
-describe('filterUserByInitials', () => {
-  test('returns users whose full_name initials matches filter excluding self', () => {
-    const user1 = eg.makeUser({ name: 'Apple', email: 'a@example.com' });
-    const user2 = eg.makeUser({ name: 'mam', email: 'f@app.com' });
-    const user3 = eg.makeUser({ name: 'app', email: 'p@p.com' });
-    const user4 = eg.makeUser({ name: 'Mobile Application', email: 'p3@p.com' });
-    const user5 = eg.makeUser({ name: 'Mac App', email: 'p@p2.com' });
-    const user6 = eg.makeUser({ name: 'app', email: 'p@p.com' });
-    const selfUser = eg.makeUser({ name: 'app', email: 'own@example.com' });
 
-    const users = deepFreeze([user1, user2, user3, user4, user5, user6, selfUser]);
+  test('returns users whose name contains diacritics but otherwise starts with filter', () => {
+    const withDiacritics = eg.makeUser({ name: 'Frödö', email: 'bagginsf@example.com' });
+    const withoutDiacritics = eg.makeUser({ name: 'Frodo', email: 'bagginz@example.com' });
+    const nonMatchingUser = eg.makeUser({ name: 'Zalix', email: 'zalix@example.com' });
+    const users = deepFreeze([withDiacritics, withoutDiacritics, nonMatchingUser]);
+    const expectedUsers = [withDiacritics, withoutDiacritics];
+    expect(filterUserStartWith(users, 'Fro', eg.makeUser().user_id)).toEqual(expectedUsers);
+  });
 
-    const expectedUsers = [user4, user5];
-    expect(filterUserByInitials(users, 'ma', selfUser.user_id)).toEqual(expectedUsers);
+  test('returns users whose name contains diacritics and filter uses diacritics', () => {
+    const withDiacritics = eg.makeUser({ name: 'Frödö', email: 'bagginsf@example.com' });
+    const withoutDiacritics = eg.makeUser({ name: 'Frodo', email: 'bagginz@example.com' });
+    const nonMatchingUser = eg.makeUser({ name: 'Zalix', email: 'zalix@example.com' });
+    const users = deepFreeze([withDiacritics, withoutDiacritics, nonMatchingUser]);
+    const expectedUsers = [withDiacritics];
```

Comment:
Let's exercise a couple more cases as mentioned here: https://github.com/zulip/zulip-mobile/pull/5292#discussion_r825098381
* (done) if you do type a diacritic, it won't match a name that has that letter without the diacritic.
* if you do type a diacritic, it won't match a name that has that letter with a different diacritic.
* if you do type a diacritic, it won't match a name that, on some other letter where the query is just plain ASCII, does have a diacritic.

Reply:
I believe these cases should be covered now.

Reply:
Thanks. This one isn't, though:

> * if you do type a diacritic, it won't match a name that, on some other letter where the query is just plain ASCII, does have a diacritic.

For example, searching for "Frödo" won't find "Frödö". This is true even though the individual pieces of the query *would* match: "Fröd" (without the last "o") would match, because it matches exactly, and "do" would match (because it matches the "dö" part with diacritics removed.)

Let's also test the corresponding cases on `filterUserThatContains` as well as `filterUserStartWith`.

Reply:
OK I think I follow you, check the latest revision.